### DISCUSSION TITLE
Best Practice for Error Handling in Serverless Functions

### DISCUSSION THREAD

**Discussion by jamesvclements** [vercel/community#396] [Help] [answered]
This goes beyond Vercel, but I'm wondering what the established best practice for handling errors in serverless functions is. 

Should you wrap the entire function in a try catch, throw custom errors, and `res` accordingly?
Should you import a bug reporting service to be notified when certain errors happen? 

Looking for the cleanest semantic option and what people have found works best. Thank you!

**Comment by samcx** [comment, marked as answer]
@jamesvclements I would personally just return the error message as the Serverless Function's response (adding a try catch inside the Serverless Function), assuming that you're doing some async action inside the Serverless Function.

I would utilize [Log Drains](https://vercel.com/integrations#logging) for tracking Function errors.
